pocket_prompt.py

```diff
@@ -87,7 +96,49 @@ def display(self):
         print(lines[0])
         print()
 
-    def prompt_filter(self):
+    def prompt_filter(self, match):
+        self.items = dict(
+            filter(
+                lambda pair: match in pair[1].given_url or match in pair[1].given_title,
+                self.items.items(),
+            )
+        )
+
+    def prompt_domains(self):
```

Comment:
The domain filtering doesn't work as expected, and there are some runtime errors as well.

Please use this as the implementation:

```python3
    def prompt_domains(self):
        domains = defaultdict(int)
        for item in self.items.values():
            domain = urlparse(item.given_url).hostname
            domains[domain] += 1
        domains = list(domains.items())

        while True:
            for idx, (domain, count) in enumerate(domains, start=1):
                print(f"{idx}. {count}: {domain}")
            try:
                domain_idx = int(input("> "))
                if 1 <= domain_idx <= len(domains):
                    filter_domain = domains[domain_idx - 1][0]
                    break
                raise ValueError
            except ValueError:
                print("Invalid index\n")
                pass

        self.items = dict(
            filter(
                lambda pair: urlparse(pair[1].given_url).hostname ==
                filter_domain,
                self.items.items(),
            ))
```

Reply:
Replaced. Thanks for the implementation :)